src/neuroconv/tools/path_expansion.py

```diff
@@ -132,3 +132,69 @@ def list_directory(self, base_directory: DirectoryPath) -> Iterable[FilePath]:
         base_directory = Path(base_directory)
         assert base_directory.is_dir(), f"The specified 'base_directory' ({base_directory}) is not a directory!"
         return (str(path.relative_to(base_directory)) for path in base_directory.rglob("*"))
+
+
+def construct_path_template(path: str, *, subject_id: str, session_id: str, **metadata_kwargs) -> str:
```

Comment:
Actually, the IBL case gives an example where `session_id` could be optional because there is another `metadata_kwarg` value in the filename that has a one-to-one correspondence with the true `session_id` value

I suppose the same point could actually be made with respect to the subject ID as well...

Perhaps the requirement should be that there is at least one keyword specified out of the important reserved words `subject_id` or `session_id`, or otherwise at least one key in `metadata_kwargs`?

Reply:
Though really this just highlights different usage between programmatic side and and GUIDE exposure

GUIDE wants this function for use in the session manager, which means it needs session and subject IDs explicitly specified in the paths